thread/thread.h

```diff
@@ -362,10 +362,30 @@ namespace photon
         {
             return waitq::wait(timeout);
         }
+        template<typename LOCK, typename PRED>
+        int wait_pred(LOCK& lock, PRED&& pred, Timeout timeout = {}) {
+            return do_wait_pred([&]{ return wait(lock, timeout); }, std::forward<PRED>(pred), timeout);
+        }
+        template<typename PRED>
+        int wait_pred_no_lock(PRED&& pred, Timeout timeout = {}) {
```

Comment:
simply name it as ```wait_no_lock```?

Reply:
change to `wait_no_lock` by using SFINAE